### PR TITLE
Final merge of proxies patchset

### PR DESCRIPTION
**Pull request by jmachowinski**

### CODE REVIEW

**Comment by jmachowinski** [comment]
There is still a problem with the metadata plugin but this looks more like a issue there as with this code.

File: lib/orogen/templates/config/Base.cmake [hunk 1]
```diff
@@ -83,6 +83,7 @@ add_dependencies(check-uptodate check-typekit-uptodate)
 # Take care of the task library
 <% if !project.self_tasks.empty? %>
 ADD_SUBDIRECTORY(${CMAKE_SOURCE_DIR}/tasks)
+ADD_SUBDIRECTORY(${CMAKE_SOURCE_DIR}/<%= Generation::AUTOMATIC_AREA_NAME %>/proxies )
```
Comment (doudou) [inline lib/orogen/templates/config/Base.cmake]: This should probably not be added if the proxies were disabled.

**Comment by jmachowinski** [comment]
Fixed.
Also added a hack for the metadata issue

**Comment by doudou** [comment]
> Also added a hack for the metadata issue

And why not, you know, fix it ?

**Comment by jmachowinski** [comment]
To quote you : 
> (4) until we get (3) widely deployed, we should simply "ignore" duplicate properties that have same name and type.

File: lib/orogen/gen/tasks.rb [hunk 2]
```diff
@@ -491,6 +491,21 @@ def generate
                         returns("int").
                         base_body("    #ifdef HAS_GETTID\nreturn syscall(SYS_gettid);\n#else\nreturn 0;\n#endif").
                         doc("returns the PID for this task")
+
```
Comment (goldhoorn) [inline lib/orogen/gen/tasks.rb]: This whole block is again (simliar to the discussion of the transformer) model knowledge and should not part of the TaskContext itself

**Comment by goldhoorn** [comment]
I rethought about this patches this and https://github.com/rock-core/drivers-orogen-transformer/pull/3.

A clear c++ model infrastructure should not cause that much of overhead. Instead of generating only the TaskProxies i vote for a structure like:

Rock::Model::Task (vote for Rock pre-namespace because it's not RTT specific anymore)
Rock::Proxy::Task

The model can held all static information that are needed like the name mapping, active plugins, typekits and so on. This needs a base class Rock::Mode::TaskExtension and a vector of extensions that are generated for this class. 

Each plugin (like the transformer) adds a extension to the c++ model which can be queried. The TaskExtension Model like Rock::Model::Transformer : public TaskExtension can be generated from the transformer plugin, and used within the rtt::TaskContext to instantiate/configure the transformer. Only the generation sequence might be tricky we have to take a detailed look.

I'm not sure at this step how we should handle the dynamic loading. I would really like to have a way to dynamically load the Model files without the need to link strictly against them. Not sure also howto handle the model representation. Maybe the best way is a singleton pattern in combination with the https://github.com/ros/class_loader we would have the need for more detailed discussions here.

**Comment by goldhoorn** [comment]
We discussed internally: This should become a plugin for orogen. Janosch rewrite this as a plugin and i support/do the needed modification for orogen if some registrations hooks missing. Therefore i close this PR.

**Pull request closed without merging**